LockboxXCUITests/LockboxXCUITests.swift

```diff
@@ -75,27 +81,106 @@ class LockboxXCUITests: BaseTestCase {
+                let uidNumber = self.getPostValues(value: uid)
+
+                self.completeVerification(uid: String(uidNumber), code: String(codeNumber)) {
+                    done()
+                }
+            } else {
                 done()
             }
         }
         task.resume()
     }
 
-    func test1LoginWithSavedLogins() {
+    private func getPostValues(value: String) -> String {
+        // From the regExp it is necessary to get only the number to add it to a json and send in POST request
+        let finalNumberIndex = value.index(value.endIndex, offsetBy: -32);
+        let numberValue = value[finalNumberIndex...];
+        return String(numberValue)
+    }
+
+    private func disconnectAccount() {
+        navigator.nowAt(Screen.LockboxMainPage)
+        navigator.performAction(Action.DisconnectFirefoxLockbox)
+        waitforExistence(app.buttons["getStarted.button"])
+        app.buttons["getStarted.button"].tap()
+        waitforExistence(app.staticTexts[emailTestAccountLogins])
+        app.webViews.links["Use a different account"].tap()
+        waitforExistence(app.webViews.textFields["Email"], timeout: 10)
+    }
+
+    private func preTestStatus() {
+        // App can start with user logged out, user logged out (email saved), user logged in
+        // if user logged in, log out
+        if (app.navigationBars["Firefox Lockbox"].exists) {
+            disconnectAccount()
+            logInAgain()
+        } else if (app.buttons["getStarted.button"].exists) {
+            // User logged out but emal remembered
+            app.buttons["getStarted.button"].tap()
+            waitforExistence(app.webViews.secureTextFields["Password"])
+            if (app.staticTexts[emailTestAccountLogins].exists) {
+                app.webViews.links["Use a different account"].tap()
+                waitforExistence(app.webViews.textFields["Email"], timeout: 10)
+                logInAgain()
+            } else {
+            // Starting like first time
+            logInAgain()
+            }
+        }
+    }
+
+    private func logInAgain() {
+        navigator.nowAt(Screen.FxASigninScreen)
+        enterDataAndTapOnSignIn()
+        checkIfAccountIsVerified()
+        waitForMainPage()
+    }
+
+    private func enterDataAndTapOnSignIn() {
+        waitforExistence(app.webViews.secureTextFields["Password"])
+        userState.fxaUsername = emailTestAccountLogins
+        navigator.performAction(Action.FxATypeEmail)
+        userState.fxaPassword = passwordTestAccountLogins
+        navigator.performAction(Action.FxATypePassword)
+
+        navigator.performAction(Action.FxATapOnSignInButton)
+        waitforExistence(app.buttons["finish.button"])
+        app.buttons["finish.button"].tap()
+        waitForMainPage()
+    }
+
+    private func waitForMainPage() {
+        waitforExistence(app.navigationBars["Firefox Lockbox"])
+        waitforExistence(app.tables.cells.staticTexts[firstEntryEmail], timeout: 10)
+    }
```

Comment:
All of this should be in the graph. 

`waitForMainPage`, `enterDataAndTapOnSignIn`, `logInAgain`, `logInAgain`, `checkIfAccountIsVerified`.

This is exactly what the graph is for.